File: remote/src/main/java/com/kelsos/mbrc/utilities/BitmapResponseBodyConverter.java
package com.kelsos.mbrc.utilities;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;

import okhttp3.ResponseBody;

import java.io.IOException;

import retrofit2.Converter;

public class BitmapResponseBodyConverter implements Converter<ResponseBody, Bitmap> {
  @Override
  public Bitmap convert(ResponseBody value) throws IOException {
    return BitmapFactory.decodeStream(value.byteStream());
  }
}

File: remote/src/main/kotlin/com/kelsos/mbrc/utilities/BitmapResponseBodyConverter.kt
package com.kelsos.mbrc.utilities

import android.graphics.Bitmap
import android.graphics.BitmapFactory
import okhttp3.ResponseBody
import retrofit2.Converter
import java.io.IOException

class BitmapResponseBodyConverter : Converter<ResponseBody, Bitmap> {
  @Throws(IOException::class)
  override fun convert(value: ResponseBody): Bitmap {
    return BitmapFactory.decodeStream(value.byteStream())
  }
}
